app/services/solr_mapper.rb

```diff
@@ -14,15 +14,85 @@ def initialize(metadata:, dataset_record_id:, dataset_record_set_id:)
   end
 
   # @return [Hash] the Solr document
+  # rubocop:disable Metrics/AbcSize
   def call
     {
       id: dataset_record_id,
-      dataset_record_set_id: dataset_record_set_id,
-      title: metadata.dig(:titles, 0, :title)
-    }
+      dataset_record_set_id_ss: dataset_record_set_id,
+      access_ssi: metadata['access'],
+      provider_ssi: metadata['provider'],
+      creators_struct_ss: metadata['creators'].to_json,
+      descriptions_tsim: retrieve_descriptions(metadata['descriptions'] || []),
+      doi_ssi: retrieve_doi(metadata['identifiers']) || '',
+      provider_identifier_ssim: map_provider_identifiers(metadata['identifiers'], metadata['provider'])
+    }.merge(map_titles(metadata['titles']))
   end
+  # rubocop:enable Metrics/AbcSize
 
   private
 
   attr_reader :metadata, :dataset_record_id, :dataset_record_set_id
+
+  # Given titles from metadata, return field value based on title type
+  def title_fields(title_type, titles)
+    titles.select { |title_obj| title_obj['title_type'] == title_type }.pluck('title')
+  end
+
+  def map_titles(titles_metadata)
```

Comment:
`titles_metadata` doesn't need to be passed as a variable since metadata is an instance variable.

Can you call this method `title_fields`? It makes it clearer that this method is producing.

Reply:
Updated method